datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaConnector.java

```diff
@@ -154,6 +156,7 @@ public synchronized void onAssignmentChange(List<DatastreamTask> tasks) {
     _logger.info("onAssignmentChange called with tasks {}", tasks);
 
     synchronized (_runningTasks) {
```

Comment:
Do you have plans to fix the metric issue in the same PR or separate PR?

Reply:
Separate PR, https://github.com/linkedin/brooklin/pull/869

Reply:
PR #869 description does not tell how it is causing the metric issue. Also, in the situation where we saw metric issue was not related to InterruptedException path. Look at this code for more ideas.

`private DatastreamTask stopTask(DatastreamTask datastreamTask, ConnectorTaskEntry connectorTaskEntry) {`